apps/web/components/PayPal/PayPalExpressButton.vue

```diff
@@ -0,0 +1,105 @@
+</template>
+
+<script setup lang="ts">
+import { OnApproveData } from '@paypal/paypal-js';
+import { orderGetters } from '@plentymarkets/plentymarkets-sdk/packages/sdk/src';
+import { v4 as uuid } from 'uuid';
+import { paypalGetters } from '~/getters/paypalGetters';
+
+const { shippingPrivacyAgreement } = useAdditionalInformation();
+const paypalButton = ref<HTMLElement | null>(null);
+const vsfCurrency = useCookie('vsf-currency').value as string;
+const fallbackCurrency = useAppConfig().fallbackCurrency as string;
+const currency = vsfCurrency?.length > 0 ? vsfCurrency : fallbackCurrency;
+
+const paypalUuid = uuid();
+const { loadScript, createTransaction, approveOrder, executeOrder } = usePayPal();
+const { createOrder } = useMakeOrder();
+const { data: cart } = useCart();
+const router = useRouter();
+const emits = defineEmits(['on-click']);
+
+const props = defineProps({
+  disabled: {
+    type: Boolean,
+  },
+});
+
+const paypal = await loadScript(currency);
+
+const onApprove = async (data: OnApproveData) => {
+  await approveOrder(data.orderID, data.payerID ?? '');
+
+  const order = await createOrder({
+    paymentId: cart.value.methodOfPaymentId,
+    shippingPrivacyHintAccepted: shippingPrivacyAgreement.value,
+  });
+
+  await executeOrder({
+    mode: 'paypal',
+    plentyOrderId: Number.parseInt(orderGetters.getId(order)),
+    paypalTransactionId: data.orderID,
+    paypalMerchantId: paypalGetters.getMerchantId() ?? '',
+  });
+
+  if (order?.order?.id) {
+    router.push('/order/success');
+  } else {
+    router.push('/order/failed');
+  }
+};
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+onMounted(() => {
+  if (paypal) {
+    const FUNDING_SOURCES = [paypal.FUNDING?.PAYPAL];
+
+    try {
+      FUNDING_SOURCES.forEach((fundingSource) => {
+        if (paypal.Buttons && fundingSource) {
+          const button = paypal.Buttons({
+            style: {
+              layout: 'vertical',
+              label: 'buynow',
+              color: 'blue',
+            },
+            fundingSource: fundingSource,
+
+            onClick() {
+              emits('on-click');
+            },
+
+            onInit(data, actions) {
+              actions.disable();
+              watch(props, (watchProps) => {
+                if (watchProps.disabled) {
+                  actions.disable();
+                } else {
+                  actions.enable();
+                }
+              });
+            },
+
+            async createOrder() {
+              const order = await createTransaction(fundingSource);
+              return order?.id ?? '';
+            },
+
+            async onApprove(data) {
+              await onApprove(data);
+            },
+          });
+
+          if (button.isEligible() && paypalButton.value) {
+            button.render('#' + paypalButton.value?.id);
+          }
+        }
+      });
+    } catch {
+      // console.error('failed to render the PayPal Buttons', error);
```

Comment:
**question**: Can we replace the commented out console error with a TODO of how we ultimately intend to handle the error?